pkg/solidity-utils/contracts/openzeppelin/MerkleProof.sol

```diff
@@ -21,13 +21,14 @@ library MerkleProof {
             if (computedHash <= proofElement) {
                 // Hash(current computed hash + current element of the proof)
                 computedHash = keccak256(abi.encodePacked(computedHash, proofElement));
+                index += 2**i;
```

Comment:
I think this breaks in the situation where there's any sorting of hashes in the upper layers. I'm pretty confident that if you tried this for larger trees then it would start marking incorrect indices as claimed in the bitmap as we'd need the hashes in each layer to coincidentally be in ascending order.

Reply:
As discussed, this does break the idea of an "index" as used in #828 however it's unique per merkle leave so is acceptable to use as a method to determine whether a particular user has claimed their distribution.

Reply:
This does have downsides for gas in comparison to explicit indices.

>One downside I see is that it will make claims with a number of participants which is just a bit over a power of 2 less efficient.
Say we have 1025 people in the tree, to balance the tree we need to pad it to 2048 which needs a bitmap 8 slots long. With explicit indices we'll fill up the first 4 slots and then the last person spills over into the next slot. With implicit indices users get placed as a near random position in those 8 slots so we'll end up initialising all of them.

That said, at worst each slot will be half filled on average so there's still significant gains.

Reply:
Right - this is not an index per se but an identifier that spans up to ```2**(Math.ceiling(Math.log2(elements)))``` which could be at worst half full

Both are wayyy better than storing claim booleans to the point where the difference between true indices vs identifiers becomes less significant compared to the other gas costs

I find this solution elegant despite the inefficiency of storing identifiers vs indexes.  I think we should do some benchmarking to put numbers to our assumptions.